src/main/scala/diplomaticobjectmodel/ConstructOM.scala

```diff
@@ -7,9 +7,14 @@ import freechips.rocketchip.diplomaticobjectmodel.logicaltree._
 import freechips.rocketchip.diplomaticobjectmodel.model.OMComponent
 import freechips.rocketchip.util.ElaborationArtefacts
 
-case object ConstructOM {
-  def constructOM(resourceBindingsMap: => ResourceBindingsMap): Unit = {
-    val om: Seq[OMComponent] = LogicalModuleTree.bind(resourceBindingsMap)
+object ConstructOM {
+  private var resourceBindingsMap: () => ResourceBindingsMap = _
+
+  def apply(rbm: () => ResourceBindingsMap): Unit = resourceBindingsMap = rbm
```

Comment:
I'd call this method `bind` or `setBindingsMap` or something more explicit than just the apply method. Usually when object have an apply method they return an instance of the companion class.

Reply:
Thanks.

Reply:
Changed to:

  def setBindingsMap(rbm: () => ResourceBindingsMap): Unit = resourceBindingsMap = rbm